rnote-engine/src/document/mod.rs

```diff
@@ -194,6 +205,26 @@ impl Document {
         self.height = new_height;
     }
 
+    pub(crate) fn expand_doc_semi_infinite_layout(&mut self, viewport: Aabb) {
+        let padding_horizontal = self.format.width * 2.0;
+        let padding_vertical = self.format.height * 2.0;
+
+        let new_viewport_bounds =
+            &viewport.extend_right_and_bottom_by(na::vector![padding_horizontal, padding_vertical]);
+
+        let new_bounds = self.bounds().merged(new_viewport_bounds);
+
+        self.x = 0.0;
+        self.y = 0.0;
+
+        if new_viewport_bounds.maxs[0] >= self.bounds().maxs[0] {
```

Comment:
what is the reason for these conditions?

Reply:
The reason for these conditions is to fix the bug with the scrollbars expanding when dragging towards the top left.
It doesn't evaluate to true when the viewport is smaller than the documents bounds.
Actually I realize now that the bug should be fixed at the root and that this is more of a workaround.
I have really no clue why but for some reason the width of the document increases when the viewport has a negative offset. This leads to self.bounds() being bigger than what it should be which in turn leads to the scroll bars bug.

Reply:
ooooh, it's the extends function that computes maxs - mins and mins is actually negative bc of the viewport...